### ISSUE TITLE
Filterfield: Accessibility improvements

### ISSUE THREAD

**Issue by tomheller** [dynatrace-oss/barista#567]
# Feature Request

## Summary

The barista filter field currently fails some accessibility tests, as it does not conform to every aspect of the [combobox role](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html).

We should look into connecting the combo box with the autocompletes via the attributes:

* `aria-expanded`
* `aria-owns`
* `aria-haspopup`

**Comment by tomheller** [comment]
Similar things as in #568 apply here. The structure of the filter field is too disconnected, with option elements not being rendered where they are expected (cdkOverlay) and this proves quite difficult to resolve without refactoring the whole component structure.
